tests/as_num.R

```diff
@@ -0,0 +1,15 @@
+library(testit)
+
+assert("Parse characters as number - class", is.numeric(json2test:::as.num("7")))
```

Comment:
I'm not very familiar with testit, does it require the namespace prefix for testing internal functions?

Reply:
i believe it does, but not 100% sure.

Reply:
@karawoo, @roninsightrx: I went down a small rabbit hole here. It actually does _not_ require triple colons [if you test through](https://www.rdocumentation.org/packages/testit/versions/0.12) `testit::test_pk()`. However, I did not have the testing structure set up to test in this format, so I have moved around some of our tests.

In the process, I realized I was actually using json2test::assert and not testit::assert. IMO this naming design choice is suboptimal, and should be addressed in a later PR.

See the new commit (which is a substantial number of LOC... apologies!).

But we can now write tests in a way much mroe similar to {testthat} and check tests with `test_pkg("json2test")` (or with `devtools::check()`)